### PR TITLE
feat: upload semcom components component

### PR DESCRIPTION
**Pull request by AbelVandenBriel**

### CODE REVIEW

**Comment by AbelVandenBriel** [comment]
Due to errors in the semcom repo this component was developed initially in the ui-transfer repo. See [PR 59](https://github.com/digita-ai/ui-transfer/pull/59), [PR 60](https://github.com/digita-ai/ui-transfer/pull/60), and [PR 64](https://github.com/digita-ai/ui-transfer/pull/64) from the ui-transfer repo.

At the time of this comment no functionality of the component was changed during the port over to this repo. Any changes are simply to accommodate strict mode being set to true in this repo, and fixing imports.

Current issues:
- Component has no width or height in the html, will look for a fix with @BelgianNoise.
- The form to upload a component needs validation, but need input from @lem-onade for that.

File: packages/semcom-components/lib/components/sem-com-register.component.ts
Comment (woutermont) [inline packages/semcom-components/lib/components/sem-com-register.component.ts:32]: Casting is normally not against strict mode ...

File: packages/semcom-components/lib/components/sem-com-register.component.ts
Comment (woutermont) [inline packages/semcom-components/lib/components/sem-com-register.component.ts:18]: No need for a type here, I think, let TS just infer it.

File: packages/semcom-components/lib/components/sem-com-register.component.ts
Comment (woutermont) [inline packages/semcom-components/lib/components/sem-com-register.component.ts:89]: This is not really needed. The surrounding layout in Figma serves as an example, but is not part of the component itself (rather it should be in the demo here)

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (woutermont) [reply packages/semcom-components/lib/components/sem-com-register.machine.ts:35]: Then the two interfaces above probably aren't neede here as well...

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (woutermont) [inline packages/semcom-components/lib/components/sem-com-register.machine.ts:290]: I know this is how it is done in NDE, but I'd prefer the Form Machines to be included in their Lit components, and when done/submitted, alert this machine by way of an event.

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (woutermont) [inline packages/semcom-components/lib/components/sem-com-register.machine.ts:301]: This is quite long. so might be beteer seperated from the config.

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (woutermont) [inline packages/semcom-components/lib/components/sem-com-register.machine.ts:348]: Same comment as in other form machine.

**Review by woutermont — CHANGES_REQUESTED** [review]
Actually I realize again that the upload component is not a semcom component ... maybe it's better if we put it in the SDK package?

It's a lot of code, so I haven't read everything in detail. Will do that after the necessary restructuring.

File: packages/semcom-components/lib/components/sem-com-register.component.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-register.component.ts:32]: What I meant with the comment is that the cast should not be necessary, but i'm doing it because strict mode causes an error here.

Without the cast, and without the any on line 18:
```
Argument of type 'Interpreter<SemComRegisterContext, any, SemComRegisterEvent, SemComRegisterState>' is not assignable to parameter of type 'ObservableInput<any>'.
  Type 'Interpreter<SemComRegisterContext, any, SemComRegisterEvent, SemComRegisterState>' is not assignable to type 'InteropObservable<any>'.
    The types of '[Symbol.observable]().subscribe' are incompatible between these types.
      Type '{ (next: (value: State<SemComRegisterContext, SemComRegisterEvent, any, SemComRegisterState>) => void, error?: ((error: any) => void) | undefined, complete?: (() => void) | undefined): Subscription; (observer: Observer<...>): Subscription; }' is not assignable to type '{ (observer?: PartialObserver<any> | undefined): Unsubscribable; (next: null | undefined, error: null | undefined, complete: () => void): Unsubscribable; (next: null | undefined, error: (error: any) => void, complete?: (() => void) | undefined): Unsubscribable; (next: (value: any) => void, error: null | undefined, c...'.
        Types of parameters 'next' and 'next' are incompatible.
          Type 'null | undefined' is not assignable to type '(value: State<SemComRegisterContext, SemComRegisterEvent, any, SemComRegisterState>) => void'.
            Type 'undefined' is not assignable to type '(value: State<SemComRegisterContext, SemComRegisterEvent, any, SemComRegisterState>) => void'.
```

File: packages/semcom-components/lib/components/sem-com-register.component.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-register.component.ts:18]: All of the comments that mention `- due to ts strict mode` need to be together for everything to work, it seems to have a cascading effect. In ui-transfer repo I do not have a type here, but strictmode causes an error on line 33 when subscribing to the state with `from(this.actor)`. If I dont have the any on the actor, and I dont cast on line 33, it doesn't work.

Error without the any but WITH the cast on line 33 (the error occurse on line 33 in the `from(this.actor)`):
```
Argument of type 'Interpreter<SemComRegisterContext, any, SemComRegisterEvent, SemComRegisterState>' is not assignable to parameter of type 'ObservableInput<any>'.
  Type 'Interpreter<SemComRegisterContext, any, SemComRegisterEvent, SemComRegisterState>' is not assignable to type 'InteropObservable<any>'.
    The types of '[Symbol.observable]().subscribe' are incompatible between these types.
      Type '{ (next: (value: State<SemComRegisterContext, SemComRegisterEvent, any, SemComRegisterState>) => void, error?: ((error: any) => void) | undefined, complete?: (() => void) | undefined): Subscription; (observer: Observer<...>): Subscription; }' is not assignable to type '{ (observer?: PartialObserver<any> | undefined): Unsubscribable; (next: null | undefined, error: null | undefined, complete: () => void): Unsubscribable; (next: null | undefined, error: (error: any) => void, complete?: (() => void) | undefined): Unsubscribable; (next: (value: any) => void, error: null | undefined, c...'.
        Types of parameters 'next' and 'next' are incompatible.
          Type 'null | undefined' is not assignable to type '(value: State<SemComRegisterContext, SemComRegisterEvent, any, SemComRegisterState>) => void'.
            Type 'undefined' is not assignable to type '(value: State<SemComRegisterContext, SemComRegisterEvent, any, SemComRegisterState>) => void'.
```

File: packages/semcom-components/lib/components/sem-com-store-selection.component.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-store-selection.component.ts:18]: the actor is passed to the necessary components in the root of component (sem-com-register.component.ts). Examples are on line 51 and 59.

File: packages/semcom-components/lib/components/sem-com-upload-form.component.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-upload-form.component.ts:38]: No, this is simply due to strict mode.

As stated in the comment  on line 19 and line 14 the actual types of the actor and formActor should not be any, but they are because the `from(this.actor)` on line 37, and the `from(this.formActor)` on line 46 and 55 cause errors when they are typed correctly. Because they are any's, if I dont type the machineState as any it won't be able to find machineState.children or machineState.matches. If the  subscribe works correctly with the `from(...)` call, a type is not necessary. It's just a temporary fix.

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-register.machine.ts:154]: Oops, forgot to add the last 2 there.

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-register.machine.ts:210]: I can, but then I cant call the event with `new NoPermissionEvent()`, I would have to call it with `SemComRegisterEvents.NO_PERMISSION`.

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-register.machine.ts:290]: The form machine is alerted by an event in the `sem-com-store-selection.component.ts` on line 115.

Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-register.machine.ts:290]: I don't know how I would include the form machine in the Lit component itself, so would need some help with that.

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-register.machine.ts:301]: Sure, I can put it in a seperate function.

File: packages/semcom-components/lib/components/sem-com-upload-form.component.ts
Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-upload-form.component.ts:15]: This component get's its actor from the root component (sem-com-register.component.ts) on lines 51 and 59.

Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-upload-form.component.ts:15]: The actor itself is defined in the sem-com-register.component.ts on line 18

File: packages/semcom-components/lib/components/sem-com-register.machine.ts
Comment (woutermont) [reply packages/semcom-components/lib/components/sem-com-register.machine.ts:210]: What? That's insane ... of course you should be able to call `new` on a class without an explicit constructor. Every class implicitly has an empty constructor.

Comment (AbelVandenBriel) [reply packages/semcom-components/lib/components/sem-com-register.machine.ts:210]: Hmm, you're correct. I must've made som other mistake.

**Comment by AbelVandenBriel** [comment]
Note: This branch had an issue which has been resolved in #175. Do not make additional comments here, and do not merge this branch. It will be closed and removed once #175 is merged.

**Pull request closed without merging**
